### ISSUE TITLE
RNIap.requestSubscription returns successful result when subscription is not confirmed (Amazon)

### ISSUE THREAD

**Issue by avoloshko** [dooboolab/react-native-iap#1222]
### Version of react-native-iap
5.2.0
### Version of react-native
0.63.3
### Platforms you faced the error (IOS or Android or both?)
Android
### Expected behavior
RNIap.requestSubscription waits for successful subscription
### Actual behavior
RNIap.requestSubscription returns successful result even though the user has not subscribed the app yet
### Tested environment (Emulator? Real Device?)
Real device, GalaxyS9, Android 10
### Steps to reproduce the behavior
- Install an app from Amazon Store
- Pick a subscription item (RNIap.requestSubscription) is called
- RNIap.requestSubscription returns the result immediately

Basically, this behavior is different from the cases when apps are installed through Google Play. Normally, RNIap.requestSubscription waits until the users subscribes (selects card and clicks OK)

**Comment by MarcosHY** [comment]
I am facing the same issue with RNIap.requestPurchase. By default (Google) it shows the pop up to cancel or confirm the purchase and in that case it returns a Json with its information. But when I use the ones from Amazon, `await RNIap.requestPurchase` automatically returns `true`. 

I have tried different configurations in Amazon App Tester but the only thing that changes is that the IAPs load or not. The app is still in Live App Testing, and the IAP is a consumable that is Live.

Using react-native 0.62.2.
Real device: Pocophone F1, Android 10
Installed from Android studio and using RNIap.setFallbackInstallSourceAndroid(RNIap.InstallSourceAndroid.AMAZON); to use Amazon's.

**Comment by hyochan** [comment]
@iaphub May you help this out?

**Comment by iaphub** [comment]
@avoloshko You're right currently the Amazon is working a little bit differently even if it is not an issue to use Amazon since you can listen to purchase & errors events.
Amazon is still in 'beta' and will definitely need some love in the coming months for all these little improvements.
I'll take the time to work on it in the future but @avoloshko @MarcosHY feel free to push a PR if you would like 👍

**Comment by dehimer** [comment]
Actual. On android request return true on moment when I just see native popup with Subscribe button.
